detox/src/realms/DetoxPrimaryContext.js

```diff
@@ -191,11 +194,18 @@ class DetoxPrimaryContext extends DetoxContext {
         await this[symbols.uninstallWorker]();
       }
     } finally {
-      if (this[_deviceAllocator]) {
-        await this[_deviceAllocator].cleanup();
-        this[_deviceAllocator] = null;
+      for (const key of Object.keys(this[_deviceAllocators])) {
+        const deviceAllocator = this[_deviceAllocators][key];
+        delete this[_deviceAllocators][key];
```

Comment:
Protect with `!== undefined`?

Reply:
I reviewed this a couple of times, but I don't see a good reason for the suggested extra caution.

I made sure that a device allocator (when created) cannot be null or anything falsy; hence, if it exists in the map, it just should be available as expected.